extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -42,27 +42,31 @@ export class AnalysesResultsManager {
     void this.logger.log('Downloading and processing all results');
 
     for (const analysis of analysisSummaries) {
-      const queryResults = await this.downloadSingleAnalysisResults(analysis.downloadLink, credentials);
-      this.analysesResults[analysis.nwo] = queryResults;
+      await this.downloadSingleAnalysisResults(analysis, credentials);
     }
   }
 
-  public getFlattenedAnalysesResults(): AnalysisResults[] {
-    return Object.entries(this.analysesResults).map(([nwo, results]) => ({ nwo, results }));
+  public getAnalysesResults(): AnalysisResults[] {
+    return [...this.analysesResults];
   }
 
   private async downloadSingleAnalysisResults(
-    downloadLink: DownloadLink,
+    analysis: AnalysisSummary,
     credentials: Credentials
-  ): Promise<QueryResult[]> {
-    const artifactPath = await downloadArtifactFromLink(credentials, downloadLink);
+  ): Promise<void> {
+    const artifactPath = await downloadArtifactFromLink(credentials, analysis.downloadLink);
+
+    let analysisResults: AnalysisResults;
 
     if (path.extname(artifactPath) === '.sarif') {
-      return await this.readResults(artifactPath);
+      const queryResults = await this.readResults(artifactPath);
+      analysisResults = { nwo: analysis.nwo, results: queryResults };
     } else {
-      // Non-problem or problem-path queries are not currently fully supported.
-      return [];
+      void this.logger.log('Non-problem or problem-path queries are not currently fully supported.');
```

Comment:
Can we simplify this message a bit?

```suggestion
      void this.logger.log('Cannot download results. Only alert and path queries are fully supported.');
```